src/quantum_computations_utilities.py

```diff
@@ -99,14 +98,32 @@ def get_prime_factors(number: int) -> List[int]:
 def compute_minimal_number_of_samples_for_desired_accuracy(outcomes_number: int, error_probability: float,
                                                            accuracy: float) -> int:
 
-    possibly_huge_number = 2 ** outcomes_number - 2
-    prime_factors_of_the_number = get_prime_factors(possibly_huge_number)
-
     samples_number = -log2(error_probability)
 
-    for prime_factor in prime_factors_of_the_number:
-        samples_number += log2(prime_factor)
+    samples_number += outcomes_number
 
     samples_number /= 2 * pow(accuracy, 2)
 
     return int(samples_number) + 1
+
+
+def compute_qft_matrix(n: int) -> ndarray:
+    """
+        Computes n x n matrix of quantum fourier transform. The formula can be found e.g. on wiki
+
+        https://en.wikipedia.org/wiki/Quantum_Fourier_transform
+
+        :param n: Dimension of the array.
+        :return: n x n ndarray of qft.
+    """
+    if n == 0:
+        return asarray([])
+    values = []
+    omega = exp(2j * pi / n)
+    for i in range(n):
```

Comment:
Hmm... This looks something like...

```
horizontal_range = numpy.tile(range(n), n).reshape(n, n)
vertical_range = horizontal_range.transpose()
full_range = horizontal_range * vertical_range
qft_matrix = numpy.power(omega, full_range) / sqrt(n)
```

Reply:
I still, as you can see, feel a little unsure about numpy's element-wise matrix operations. But yea, it looks exactly like what I need.